### ISSUE TITLE
Ultralytics does not support the inference using the TensorRT format files of rtdetr.

### ISSUE THREAD

**Issue by mo-lx** [ultralytics/ultralytics#22747]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

<img width="844" height="133" alt="Image" src="https://github.com/user-attachments/assets/7841d0af-54e6-4893-b1ea-088df7da99c9" />

### Environment

Package                Version
---------------------- --------------------
addict                 2.4.0
albucore               0.0.33
albumentations         1.4.11
aliyun-python-sdk-core 2.16.0
aliyun-python-sdk-kms  2.16.5
annotated-types        0.7.0
appdirs                1.4.4
certifi                2025.7.14
cffi                   2.0.0
charset-normalizer     3.4.2
click                  8.1.8
colorama               0.4.6
coloredlogs            15.0.1
contourpy              1.3.0
crcmod                 1.7
cryptography           46.0.1
cuda-toolkit           13.0.1
cycler                 0.12.1
dill                   0.3.8
efficientnet_pytorch   0.7.1
einops                 0.8.1
eval_type_backport     0.2.2
filelock               3.14.0
flatbuffers            25.2.10
fonttools              4.59.0
fsspec                 2025.7.0
grad-cam               1.5.4
huggingface-hub        0.35.0
humanfriendly          10.0
idna                   3.10
imageio                2.37.0
importlib_metadata     8.7.0
importlib_resources    6.5.2
Jinja2                 3.1.6
jmespath               0.10.0
joblib                 1.5.2
kiwisolver             1.4.7
lazy_loader            0.4
Mako                   1.3.10
Markdown               3.9
markdown-it-py         3.0.0
MarkupSafe             3.0.2
matplotlib             3.9.4
mdurl                  0.1.2
mmengine               0.10.7
model-index            0.1.11
mpmath                 1.3.0
networkx               3.2.1
numpy                  2.0.2
nvidia-cuda-runtime    13.0.88
onnx                   1.18.0
onnxruntime            1.19.2
opencv-python          4.12.0.88
opencv-python-headless 4.12.0.88
opendatalab            0.0.10
openmim                0.3.9
openxlab               0.1.2
ordered-set            4.1.0
oss2                   2.17.0
packaging              24.2
pandas                 2.3.1
pillow                 11.3.0
pip                    25.1.1
platformdirs           4.4.0
prettytable            3.16.0
protobuf               6.31.1
psutil                 7.0.0
py-cpuinfo             9.0.0
pycocotools            2.0.10
pycparser              2.23
pycryptodome           3.23.0
pycuda                 2025.1.2
pydantic               2.11.9
pydantic_core          2.33.2
Pygments               2.19.2
pyparsing              3.2.3
pyreadline3            3.5.4
python-dateutil        2.9.0.post0
pytools                2024.1.14
pytorch-wavelets       1.3.0
pytz                   2023.4
PyWavelets             1.6.0
pywin32                311
PyYAML                 6.0.2
regex                  2025.9.18
requests               2.28.2
rich                   13.4.2
safetensors            0.6.2
scikit-image           0.24.0
scikit-learn           1.6.1
scipy                  1.13.1
seaborn                0.13.2
setuptools             60.2.0
simsimd                6.5.3
six                    1.17.0
stringzilla            4.0.13
sympy                  1.14.0
tabulate               0.9.0
tensorrt               10.14.1.48.post1
tensorrt-cu13          10.14.1.48.post1
tensorrt_cu13_bindings 10.14.1.48.post1
tensorrt_cu13_libs     10.14.1.48.post1
termcolor              3.1.0
thop                   0.1.1.post2209072238
threadpoolctl          3.6.0
tidecv                 1.0.1
tifffile               2024.8.30
timm                   1.0.7
tomli                  2.2.1
torch                  2.7.1+cu118
torch-dct              0.1.6
torchaudio             2.7.1+cu118
torchvision            0.22.1+cu118
tqdm                   4.65.2
ttach                  0.0.3
typing_extensions      4.14.1
typing-inspection      0.4.1
tzdata                 2025.2
ultralytics            8.3.168
ultralytics-thop       2.0.14
urllib3                1.26.20
wcwidth                0.2.13
wheel                  0.45.1
yapf                   0.43.0
zipp                   3.23.0

### Minimal Reproducible Example

...

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @mo-lx, thank you for opening this issue and for your interest in Ultralytics 🚀! This is an automated response to help gather initial details; an Ultralytics engineer will review your report and assist you further soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples. Many common questions and known limitations around export and inference formats (including TensorRT) are documented there 💡.

Since this appears to be a 🐛 Bug Report related to RT-DETR TensorRT inference, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) (MRE) to help us debug it. For example, it would be very helpful if you could include:
- The exact command(s) you use to export RT-DETR to TensorRT
- The exact command(s) or code you use for inference with the exported engine
- The full traceback or error message (pasted as text, not only a screenshot if possible)
- Details on the model you are using (RT-DETR variant, custom vs. pretrained, etc.)

If this is instead a custom training ❓ Question about RT-DETR or TensorRT workflows, please provide as much information as possible, including dataset image examples (if relevant) and training/export logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best for additional help and discussion:
- For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧
- Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/)
- Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community 😄

## Upgrade

Before we dig deeper, please ensure you have upgraded to the latest `ultralytics` package and that all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) are satisfied in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). This helps verify the issue is not already resolved in a newer release:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Once you share the MRE and additional details, it will greatly speed up investigation by the engineering team.

**Comment by Y-T-G** [comment]
Works fine:

```python
In [2]: from ultralytics import RTDETR

In [3]: model = RTDETR("weights/rtdetr-l.engine")

In [4]: results = model()
WARNING ⚠️ 'source' is missing. Using 'source=/ultralytics/ultralytics/assets'.
Loading weights/rtdetr-l.engine for TensorRT inference...
[11/21/2025-22:53:36] [TRT] [I] Loaded engine size: 124 MiB
[11/21/2025-22:53:36] [TRT] [I] [MS] Running engine with multi stream info
[11/21/2025-22:53:36] [TRT] [I] [MS] Number of aux streams is 2
[11/21/2025-22:53:36] [TRT] [I] [MS] Number of total worker streams is 3
[11/21/2025-22:53:36] [TRT] [I] [MS] The main stream provided by execute/enqueue calls is the first worker stream
[11/21/2025-22:53:36] [TRT] [I] [MemUsageChange] TensorRT-managed allocation in IExecutionContext creation: CPU +0, GPU +112, now: CPU 0, GPU 235 (MiB)

image 1/2 /ultralytics/ultralytics/assets/bus.jpg: 640x640 4 persons, 1 bus, 1 traffic light, 1 fire hydrant, 2 ties, 5.1ms
image 2/2 /ultralytics/ultralytics/assets/zidane.jpg: 640x640 2 persons, 2 ties, 5.1ms
Speed: 2.0ms preprocess, 5.1ms inference, 0.9ms postprocess per image at shape (1, 3, 640, 640)
```

**Comment by Y-T-G** [comment]
You're using an outdated local clone of Ultralytics. RTDETR supports TensorRT inference since 8.1.26, which was released over 20 months ago.
